src/Controls/tests/UITests/Tests/_ViewUITests.cs

```diff
@@ -13,14 +13,40 @@ public abstract class _ViewUITests : UITestBase
 
 		protected override void FixtureSetup()
 		{
-			base.FixtureSetup();
-			NavigateToGallery();
+			int retries = 0;
+			while(true)
+			{
+				try
+				{
+					base.FixtureSetup();
+					NavigateToGallery();
+					break;
+				}
+				catch (Exception)
+				{
+					if (retries++ < 1)
+					{
+						TestContext.Error.WriteLine("There was a problem with the FixtureSetup retrying");
+						Reset();
+					}
+					else
+					{
+						throw;
+					}
+				}
+			}
 		}
 
 		protected override void FixtureTeardown()
 		{
 			base.FixtureTeardown();
-			App.NavigateBack();
+			try
+			{
+				App.NavigateBack();
+			}
+			catch (Exception)
```

Comment:
Would it make sense to have the retry logic here as will as inside FixtureSetup? 

If `NavigateBack` fails then we run `SaveDiagnosticLogs()` to track why the `teardown` seemed to have failed? 

This will also associate the failure with the correct test right? If the retry logic is inside `FixtureSetup` then the diagnostics will get saved with the next test vs the test that left us in a weird state yea?

Reply:
I don't think it is necessary here, if this fails and there isn't another test afterwards we don't really want to restart it. If it fails and we run another test then Setup will fail and do the restart there.

I don't think we need to add SaveDiagnosticLogs to this, instead I think logging an error message would be better.